datamap/example/src/main/java/org/apache/carbondata/datamap/minmax/MinMaxDataMapFactory.java

```diff
@@ -0,0 +1,365 @@
+    if (shardPaths == null) {
+      String dataMapStorePath = DataMapWriter.getDefaultDataMapPath(
+          getCarbonTable().getTablePath(), segment.getSegmentNo(), dataMapName);
+      CarbonFile[] carbonFiles = FileFactory.getCarbonFile(dataMapStorePath).listFiles();
+      shardPaths = new HashSet<>();
+      for (CarbonFile carbonFile : carbonFiles) {
+        shardPaths.add(carbonFile.getAbsolutePath());
+      }
+      segmentMap.put(segment.getSegmentNo(), shardPaths);
+    }
+
+    for (String shard : shardPaths) {
+      MinMaxIndexDataMap dataMap = new MinMaxIndexDataMap();
+      dataMap.init(new MinMaxDataMapModel(shard, cache, segment.getConfiguration()));
+      dataMap.initOthers(getCarbonTable(), dataMapMeta.getIndexedColumns());
+      dataMaps.add(dataMap);
+    }
+    return dataMaps;
+  }
+
+  @Override
+  public DataMapMeta getMeta() {
+    return this.dataMapMeta;
+  }
+
+  @Override
+  public DataMapLevel getDataMapLevel() {
+    return DataMapLevel.CG;
+  }
+
+  @Override
+  public List<CoarseGrainDataMap> getDataMaps(DataMapDistributable distributable)
+      throws IOException {
+    List<CoarseGrainDataMap> coarseGrainDataMaps = new ArrayList<>();
+    MinMaxIndexDataMap minMaxIndexDataMap = new MinMaxIndexDataMap();
+    String indexPath = ((MinMaxDataMapDistributable) distributable).getIndexPath();
+    minMaxIndexDataMap.init(
+        new MinMaxDataMapModel(indexPath, cache, FileFactory.getConfiguration()));
+    minMaxIndexDataMap.initOthers(getCarbonTable(), dataMapMeta.getIndexedColumns());
+    coarseGrainDataMaps.add(minMaxIndexDataMap);
+    return coarseGrainDataMaps;
+  }
+
+  /**
+   * returns all the directories of lucene index files for query
+   * Note: copied from BloomFilterDataMapFactory, will extract to a common interface
+   */
+  private CarbonFile[] getAllIndexDirs(String tablePath, String segmentId) {
+    List<CarbonFile> indexDirs = new ArrayList<>();
+    List<TableDataMap> dataMaps;
+    try {
+      // there can be multiple bloom datamaps present on a table, so get all datamaps and form
+      // the path till the index file directories in all datamaps folders present in each segment
+      dataMaps = DataMapStoreManager.getInstance().getAllDataMap(getCarbonTable());
+    } catch (IOException ex) {
+      LOGGER.error(String.format(
+          "failed to get datamaps for tablePath %s, segmentId %s", tablePath, segmentId), ex);
+      throw new RuntimeException(ex);
+    }
+    if (dataMaps.size() > 0) {
+      for (TableDataMap dataMap : dataMaps) {
+        if (dataMap.getDataMapSchema().getDataMapName().equals(this.dataMapName)) {
+          List<CarbonFile> indexFiles;
+          String dmPath = CarbonTablePath.getDataMapStorePath(tablePath, segmentId,
+              dataMap.getDataMapSchema().getDataMapName());
+          FileFactory.FileType fileType = FileFactory.getFileType(dmPath);
+          final CarbonFile dirPath = FileFactory.getCarbonFile(dmPath, fileType);
```

Comment:
please use org.apache.carbondata.core.datastore.impl.FileFactory#getCarbonFile(java.lang.String) this method. This method internally takes the fileType also.

Reply:
OK. And we can mark this method as deprecated later